### ISSUE TITLE
slts failing on main 

### ISSUE THREAD

**Issue by universalmind303** [apache/arrow-datafusion#9467]
### Describe the bug

a bunch of casts from floats to ints don't seem to be working in the slts

```diff
-   NULL 10 1
+   NULL 1E+1 1
```

### To Reproduce

```sh
> git checkout main
> git pull
> cargo test -p datafusion-sqllogictest   
```

### Expected behavior

tests pass on main

**Comment by comphead** [comment]
on my local tests are okay... but PR is failing

**Comment by comphead** [comment]
https://github.com/apache/arrow-datafusion/issues/9464 Maybe this related
all failures looks like on number representation

**Comment by comphead** [comment]
Fixed via #9471
